.circleci/config.yml

```diff
@@ -13,13 +13,34 @@ executors:
           POSTGRES_USER: postgres
           POSTGRES_PASSWORD: secretpass
           POSTGRES_DB: ttasmarthub
+commands:
+  create_combined_yarnlock:
+    description: "Concatenate all yarn.json files into single file.
+      File is used as checksum source for part of caching key."
+    parameters:
+      filename:
+        type: string
+        default: "combined-yarnlock.txt"
+    steps:
+      - run:
+          name: Combine package-lock.json files to single file
+          command: cat yarn.lock frontend/yarn.lock > << parameters.filename >>
 jobs:
   build:
-    # Each job must declare an executor
     executor: docker-executor
     steps:
       - checkout
+      - create_combined_yarnlock
+      - restore_cache:
+          keys:
+            - v1-yarn-deps-{{ checksum "combined-yarnlock.txt" }}
+            # If no exact match is found, restore partial cache
```

Comment:
`v1-yarn-deps-` confused me for a second, from reading the docs appears to be there case we want to manually bust the cache. Can you add something along those lines to this comment? (or correct me if I am wrong!)

Reply:
Making sure you and I are on the same page; sorry if you already knew this:
If the checksum did change (because we added a new module to yarn.lock), then the second entry in this list, `v1-yarn-deps-`, is used to try to look up a cache.  This allows circle ci to find the most recently generated cache used from any branch.  This is old/other branch list of dependancies is what is restored (this is a partial cache restore). Then, when a dependency tool runs, it would discover outdated dependencies and update them.  **_Was that clear from the comment in the config file or should I add more?_**

To manually bust the cache, we would bump the version like
```
- v2-yarn-deps-{{ checksum "combined-yarnlock.txt" }}
- v2-yarn-deps-
```
I'll add a new comment about that.

Reply:
Oh I think I get it now... Thanks for explaining!